src/source/native.rs

```diff
@@ -57,221 +47,130 @@ impl Default for NativeServerConfig {
     }
 }
 
-impl NativeServer {
-    /// Create a new NativeServer
-    pub fn new(
-        chans: Vec<hopper::Sender<metric::Event>>,
-        config: NativeServerConfig,
-    ) -> NativeServer {
-        NativeServer {
-            chans: chans,
-            ip: config.ip,
-            port: config.port,
-            tags: config.tags,
+impl From<NativeServerConfig> for TCPConfig {
+    fn from(item: NativeServerConfig) -> Self {
+        TCPConfig {
+            host: item.ip,
```

Comment:
@blt Technically speaking there is no difference (short of some field names) between NativeServerConfig/GraphiteConfig and TCPConfig.

Should we go through the exercise of porting these types now?

Reply:
Hmm, you know, that's a good question. I don't have an answer but I bet we'll know with hindsight. Maybe let's make an issue for 1.0 and see how we feel then?